CHANGES.md

```diff
@@ -2,28 +2,39 @@
 
 ## Breaking change
 
-* `Lexer` now takes a lifetime `'input` which allows the input to last longer
-  than the `Lexer` itself. `Lexer::span_str` and `Lexer::span_lines_str` have
-  changed from:
+* The `Lexer` trait has been broken into two: `Lexer` and `NonStreamingLexer`.
+  The former trait is now only capable of producing `Lexeme`s: the latter is
+  capable of producing substrings of the input and calculating line/column
+  information. This split allows the flexibility to introduce streaming lexers
+  in the future (which will not be able to produce substrings of the input in
+  the same way as a `NonStreamingLexer`).
+
+  Most users will need to replace references to the `Lexer` trait
+  in their code to `NonStreamingLexer`.
+
+* `NonStreamingLexer` takes a lifetime `'input` which allows the input to last
+  longer than the `NonStreamingLexer` itself. `Lexer::span_str` and
+  `Lexer::span_lines_str` had the following definitions:
     ```rustc
     fn span_str(&self, span: Span) -> &str;
     fn span_lines_str(&self, span: Span) -> &str;
     ```
-  to:
+  As part of `NonStreamingLexer` their definitions are now:
     ```rustc
     fn span_str(&self, span: Span) -> &'input str;
     fn span_lines_str(&self, span: Span) -> &'input str;
     ```
-  This change allows users to throw away the `Lexer` but still keep around
-  structures (e.g. ASTs) which reference the user's input.
+  as part of `NonStreamingLexer`. This change allows users to throw away the
```

Comment:
`as part of NonStreamingLexer.` not needed?

Reply:
Fixed in https://github.com/softdevteam/grmtools/pull/178/commits/926827a0c5154c1fbb0ddd3c67518fb4d9a14d89.